test/php/library/Icinga/Crypt/RSATest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+// Icinga Web 2 | (c) 2020 Icinga Development Team | GPLv2+
+
+namespace Tests\Icinga\Crypt;
+
+use Icinga\Crypt\RSA;
+use Icinga\Test\BaseTestCase;
+use InvalidArgumentException;
+use UnexpectedValueException;
+
+class RSATest extends BaseTestCase
+{
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    public function testLoadKeyThrowsExceptionIfMoreThanTwoKeysGiven()
+    {
+        (new RSA())->loadKey('one','two','three');
+    }
+
+    /**
+     * @expectedException UnexpectedValueException
+     */
+    public function testGetPublicKeyThrowsExceptionIfNoPublicKeySet()
+    {
+        (new RSA())->getPublicKey();
+    }
+
+    /**
+     * @expectedException UnexpectedValueException
+     */
+    public function testGetPrivateKeyThrowsExceptionIfNoPrivateKeySet()
+    {
+        (new RSA())->getPrivateKey();
+    }
+    public function testLoadKeyAutomaticallyDetectsThePublicAndPrivateKey()
+    {
+        list($privateKey, $publicKey) = RSA::keygen();
+
+        $rsa = (new RSA())->loadKey($publicKey, $privateKey);
+        $this->assertSame($privateKey, $rsa->getPrivateKey());
+        $this->assertSame($publicKey, $rsa->getPublicKey());
+
+        $rsa = (new RSA())->loadKey($privateKey, $publicKey);
+        $this->assertSame($privateKey, $rsa->getPrivateKey());
+        $this->assertSame($publicKey, $rsa->getPublicKey());
+    }
+
+    public function testEncryptReturnsEmptyArrayIfNoArgumentsGiven()
```

Comment:
Please also test the other 3 functions for the empty array return (in new test methods).